### DISCUSSION TITLE
fo-dicom

### DISCUSSION THREAD

**Discussion by bec1442** [fo-dicom/fo-dicom#1638] [Q&A]
I have a dicom tag (300a,012c) that stores the isocenter position as three numbers separated by "\\" (6.23769738348594\\-0.74371947\\-16.544515859162)  and this is the way I am trying to extract the value
var isocenter = dicomDataset.GetSequence(DicomTag.BeamSequence).Items[0].GetSequence(DicomTag.ControlPointSequence).Items[0].GetSingleValueOrDefault<string>(DicomTag.IsocenterPosition, "");
doesn't retune anything.  The code works if I retrieve another value in the same sequence (e.g. BeamLimitingDeviceAngle).
I believe it is due to the way the data is formatted.
Any advice how to get this x\y\z   ?

**Comment by mrbean-bremen** [comment]
As you expect an array of (3) values, you would better use [GetValues](https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.DicomDataset.html#FellowOakDicom_DicomDataset_GetValues__1_FellowOakDicom_DicomTag_) - as far as I remember, `GetSingleValueOrDefault` does not work for multi-valued attributes, e.g. something like
```cs
var isocenterXYZArray = dicomDataset.GetSequence(DicomTag.BeamSequence).Items[0]
    .GetSequence(DicomTag.ControlPointSequence).Items[0].GetValues<double>(DicomTag.IsocenterPosition);
```
should work.
If you are not sure that the tag exists, you could use [TryGetValues](https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.DicomDataset.html#FellowOakDicom_DicomDataset_TryGetValues__1_FellowOakDicom_DicomTag___0____) instead.

**Reply by bec1442** [reply]
Many thanks for the fast reply mrbeans.
I did try your modified code but returns “System.Double[]” instead of actuals.
And yes, the tag exists.

**Reply by bec1442** [reply]
My bad, it is all good
The following works 
double[] isocenterXYZArray = dicomDataset.GetSequence(DicomTag.BeamSequence).Items[0].GetSequence(DicomTag.ControlPointSequence).Items[0].GetValues<double>(DicomTag.IsocenterPosition);

many thanks
